File: bridges/voiceover/Sources/VoiceOverBridgeDomain/Ports/Transcript.swift
// ROLE: port -- the human-readable record of everything a session did.
// IMPLEMENTED BY: FileTranscript, over the FileWriter seam; FakeTranscript.
// USED BY: the Session controller, and the Hello handler, which hands `logPath` to the agent.
// Nothing here throws: an implementation swallows its own IO failures, so a broken log never stops a teardown.
public protocol Transcript: AnyObject {
	var logPath: String { get }

	func open()

	func sessionOpened(mode: String, voice: String, persona: String)

	/// Written on the capture thread, so an implementation must not block on anything but its own append.
	func speech(_ text: String)

	/// Called before the command is sent, so a command that hung stays in the record.
	func gesture(_ command: String)

	/// Records only the length, never the text: typed text may be a secret.
	func typed(_ length: Int)

	func announced(_ text: String)

	func note(_ text: String)

	func sessionClosed(reason: String)
}

File: bridges/nvda/addon/globalPlugins/nvdaMcpBridge/domain/ports/transcript.py
# nvdaMcpBridge domain -- the Transcript port.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# ROLE: domain port; the human-readable record of everything a session did.
# USED BY: the Session controller.
# IMPLEMENTED BY: adapters/file_transcript.py and tests/fakes/transcript.py.

from __future__ import annotations

from abc import ABC, abstractmethod


class Transcript(ABC):
	@property
	@abstractmethod
	def path(self) -> str:
		pass

	@abstractmethod
	def open(self) -> None: ...

	@abstractmethod
	def session_opened(self, mode: str, synth: str, persona: str) -> None:
		"""persona is empty when the server declared none."""

	@abstractmethod
	def gesture(self, gesture_id: str) -> None: ...

	@abstractmethod
	def typed(self, length: int) -> None:
		"""Record only the length: the typed text may be a secret and must never reach disk."""

	@abstractmethod
	def speech(self, text: str) -> None: ...

	@abstractmethod
	def note(self, text: str) -> None: ...

	@abstractmethod
	def session_closed(self, reason: str) -> None: ...
